src/components/task-list/task-list.component.js

```diff
@@ -1,26 +1,42 @@
-import { useState } from "react";
 import styles from "./task-list.module.css";
 
 const TaskList = (props) => {
-  const [completed, setCompleted] = useState([]);
-  const tasks = props.tasks;
+  const { tasks, taskCompletionToggle, handleTaskCompleted } = props;
 
-  const toggleCompletion = (event) => {
-    setCompleted({ ...completed });
+  const completionToggle = (event) => {
+    taskCompletionToggle(event);
+  };
+
+  const clickCheckbox = (clickEvent) => {
+    handleTaskCompleted(clickEvent);
   };
 
   return (
     <ul>
       {tasks.map((task, index) => {
         return (
-          <li className={styles.task} key={index}>
+          <li
+            className={task.completed ? styles.completedTask : styles.task}
+            key={index}
```

Comment:
you can use the `index` here, or better yet the new `task.id`

Reply:
Now using the task.id as a key for the <li>.